src/cmdstan/command_helper.hpp

```diff
@@ -363,74 +468,75 @@ void get_theta_hat_csv(const std::string &fname,
+          << "\",  expecting param \"" << cparam_names[i] << "\", found \""
           << names[i + col_offset] << "\"" << std::endl;
       throw std::invalid_argument(msg.str());
     }
     try {
-      theta_hat[i] = std::stod(values[i + col_offset]);
+      cparams[i] = std::stod(values[i + col_offset]);
     } catch (const std::exception &e) {
-      msg << "Error parsing CSV file, bad value for " << param_names[i]
+      msg << "Error parsing CSV file, bad value for " << cparam_names[i]
           << std::endl;
       throw std::invalid_argument(msg.str());
     }
   }
+  std::vector<double> uparams = unconstrain_params(model, cparams);
+  return Eigen::Map<Eigen::VectorXd>(&uparams[0], uparams.size());
 }
 
 /**
- * Parse a JSON file and extract the model parameter estimates.
+ * Parse a JSON file of a set of parameter estimates on the
+ * constrained scale and unconstrain them.
  * Helper function get_var_context throws exception if cannot
  * open or parse JSON file.
  *
- * @param fname name of file which exists and has read perms.
- * @param param_names vector of model constrained parameter names.
- * @param theta_hat Eigen vector for estimated parameters
+ * @param fname name of file which exists and has read perms
+ * @param model Stan model
+ * @return Eigen vector of unconstrained parameter estimates
  */
-void get_theta_hat_json(const std::string &fname,
-                        const std::vector<std::string> &param_names,
-                        Eigen::VectorXd &theta_hat) {
-  std::vector<std::string> var_names;
-  std::vector<std::string> splits;
-  std::string cur_name = "";
-  for (auto &&param_name : param_names) {
-    boost::algorithm::split(splits, param_name, boost::is_any_of("."),
-                            boost::token_compress_on);
-    if (cur_name.compare(splits[0]) != 0)
-      var_names.push_back(splits[0]);
-    cur_name = splits[0];
-  }
+Eigen::VectorXd get_laplace_mode_json(const std::string &fname,
+                                      const stan::model::model_base &model) {
+  std::stringstream msg;
+  std::vector<std::string> param_names;
+  model.get_param_names(param_names);
+  std::vector<std::string> cparam_names;
+  model.constrained_param_names(cparam_names);
+  std::vector<double> cparams(cparam_names.size());
 
   std::shared_ptr<stan::io::var_context> context = get_var_context(fname);
   Eigen::Index offset = 0;
-  for (auto &&var_name : var_names) {
-    const auto param_vec = context->vals_r(var_name);
-    for (Eigen::Index i = 0; i < param_vec.size(); ++i) {
-      theta_hat[offset] = param_vec[i];
+  for (auto &&param_name : param_names) {
+    const auto param_vec = context->vals_r(param_name);
+    for (size_t i = 0; i < param_vec.size(); ++i) {
+      cparams[offset] = param_vec[i];
       ++offset;
     }
   }
+  std::vector<double> uparams = unconstrain_params(model, cparams);
+  return Eigen::Map<Eigen::VectorXd>(&uparams[0], uparams.size());
```

Comment:
Same question as above

Reply:
not using map - allocate and copy.   we're not coding any of this for efficiency